src/clients/github.js

```diff
@@ -0,0 +1,280 @@
+    return Promise.reject(new EmptyGistError());
+  }
+
+  const response =
+    await _performWithRetryNetworkErrors(() => github.getGist().create(gist));
+
+  const gistData = response.data;
+  if (_canUpdateGist(user)) {
+    return _updateGistWithImportUrl(github, gistData);
+  }
+  return gistData;
+}
+
+async function loadFromId(gistId, user) {
+  const github = _clientForUser(user);
+  const gist = github.getGist(gistId);
+  const response =
+    await _performWithRetryNetworkErrors(() => gist.read(), {retries: 3});
+  return response.data;
+}
+
+function _buildGistFromProject(project) {
+  const files = {};
+  if (trim(project.sources.html)) {
+    files['index.html'] = {
+      content: project.sources.html,
+      language: 'HTML',
+    };
+  }
+  if (trim(project.sources.css)) {
+    files['styles.css'] = {
+      content: project.sources.css,
+      language: 'CSS',
+    };
+  }
+  if (trim(project.sources.javascript)) {
+    files['script.js'] = {
+      content: project.sources.javascript,
+      language: 'JavaScript',
+    };
+  }
+  if (project.enabledLibraries.length || project.hiddenUIComponents.length) {
+    files['popcode.json'] = {
+      content: _createPopcodeJson(project),
+      language: 'JSON',
+    };
+  }
+
+  return {
+    description: 'Exported from Popcode.',
+    'public': true,
+    files,
+  };
+}
+
+async function _updateGistWithImportUrl(github, gistData) {
+  const gist = github.getGist(gistData.id);
+  const uri = document.createElement('a');
+  uri.setAttribute('href', '/');
+  uri.search = `gist=${gistData.id}`;
+
+  const description = `${gistData.description} Click to import: ${uri.href}`;
+  const response =
+    await _performWithRetryNetworkErrors(() => gist.update({description}));
+  return response.data;
+}
+
+function _createPopcodeJson(project) {
+  const json = {};
+  if (project.enabledLibraries.length) {
+    json.enabledLibraries = project.enabledLibraries;
+  }
+  if (project.hiddenUIComponents.length) {
+    json.hiddenUIComponents = project.hiddenUIComponents;
+  }
+  return JSON.stringify(json);
+}
+
+function _performWithRetryNetworkErrors(perform, options = {}) {
+  return performWithRetries(
+    perform,
+    errorMessage => errorMessage === NETWORK_ERROR,
+    options,
+  );
+}
+
+function _performWithRetryNotFoundOrNetworkErrors(perform) {
+  const retryErrors = [NETWORK_ERROR, 'NOT_FOUND'];
+  return performWithRetries(
+    perform,
+    errorMessage => retryErrors.includes(errorMessage),
+  );
+}
+
+async function _createSourceFile(
+  github, userName, repoName, branchName, fileName, source) {
+  return github.getRepo(userName, repoName).
+    writeFile(
+      branchName,
+      fileName,
+      source,
+      COMMIT_MESSAGE,
+      {},
+    );
+}
+
+async function _createHtmlFile(github, project, userName, repoName) {
+  if (trim(project.sources.html)) {
+    return _createSourceFile(
+      github,
+      userName,
+      repoName,
+      MASTER,
+      'raw_index.html',
```

Comment:
Vestigial from when we created the raw_index.html and index.html (preview) on the same branch. Fixed